libraries/SITL/SIM_Precland.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "SIM_Precland.h"
+#include "AP_HAL/AP_HAL.h"
+#include "AP_Math/AP_Math.h"
+#include "AP_Common/Location.h"
+#include <stdio.h>
+
+using namespace SITL;
+
+// table of user settable parameters
+const AP_Param::GroupInfo SIM_Precland::var_info[] = {
+
+    // @Param: ENABLE
```

Comment:
Should we just use the `TYPE` parameter instead with `0` meaning "none"?

Reply:
I prefer to keep ENABLE on all sitl object from a control point of view, it is simplier to enable/disable .